cdap-app-fabric/src/main/java/co/cask/cdap/app/runtime/ProgramController.java

```diff
@@ -116,9 +117,24 @@ public boolean isDone() {
     }
   }
 
+  /**
+   * Returns the program Id which this controller is controlling.
+   */
+  Id.Program getProgramId();
+
+  /**
+   * Returns the run Id which this controller is controlling.
+   */
   RunId getRunId();
 
   /**
+   * Returns the component name within a given program which this controller is controlling
+   * or {@code null} if there is no component name for this controller.
+   */
+  @Nullable
+  String getComponentName();
+
```

Comment:
This API is internal and component is for the case of controller component run of a program (e.g. Flowlet). The same terminology is being used in runtime service and program runner.